core/stream/convert.go

```diff
@@ -113,6 +113,11 @@ func resolveImplicitMappings(count int, mappings []mapping, srcFmt *Format, srcD
 			} else if srcFmt.HasColorComponent() {
 				m.src = buf0
 			}
+		case Channel_Luminance:
+			if c := srcFmt.HasSingleColorComponent(); c != nil {
+				m.src = buf{srcData, c, srcFmt.BitOffsets()[c], uint32(srcFmt.Stride()) * 8}
+			}
+			// TODO: RGB->Luminance conversion
```

Comment:
Make it clearer that we're just reusing the first color component we find for luminance. A link to a github issue would also be good.